Automattic/woocommerce-payments
# fix: tokenized cart support for special characters and local pickup

Fixes #

#### Changes proposed in this Pull Request









The special characters aren't decoded in the shipping options.
I also added some additional information about the pickup location in the PRB dialog.

Before:
<img width="594" alt="Screenshot 2024-07-17 at 6 49 40 PM" src="https://github.com/user-attachments/assets/2263c8f9-9d89-42b6-897a-c5e0df89f65a">

After:
<img width="603" alt="Screenshot 2024-07-17 at 6 50 09 PM" src="https://github.com/user-attachments/assets/48502144-9b39-4778-a8fb-24b37d8730ce">

As a reference, this is how the options are displayed on the checkout block:
<img width="705" alt="Screenshot 2024-07-18 at 9 22 15 AM" src="https://github.com/user-attachments/assets/946b891e-7edb-4b9b-886c-afb5212a7bbe">

#### Testing instructions





- Enable the "Enable Cart-Token implementation for PRBs" flag in your dev tools (you'll need to update them)
- Ensure you have Google Pay/Apple Pay enabled in the merchant's settings
- Ensure you have "Local pickup" locations configured and enabled (you'll need your checkout page to use WooCommerce Blocks)
<img width="1097" alt="Screenshot 2024-07-17 at 6 49 45 PM" src="https://github.com/user-attachments/assets/b9720be2-585e-4aa7-9b86-194685554111">
- Go to a physical product
- Click on the GooglePay/ApplePay button
- The shipping options should have decoded special characters and provide some information about the address pickup


-------------------


**Post merge**
